File: backend/app/schemas/attachment.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class AttachmentOut(BaseModel):
    id: uuid.UUID
    entity_type: str
    entity_id: uuid.UUID
    file_name: str
    file_url: str
    file_size: int | None
    mime_type: str | None
    uploaded_by: uuid.UUID | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/attachment.ts
import request from './request'

export interface Attachment {
  id: string
  entity_type: string
  entity_id: string
  file_name: string
  file_url: string
  file_size: number | null
  mime_type: string | null
  uploaded_by: string | null
  created_at: string
}

export function getAttachments(entity_type: string, entity_id: string) {
  return request.get<Attachment[]>('/attachments', { params: { entity_type, entity_id } })
}

export function uploadAttachment(entity_type: string, entity_id: string, file: File) {
  const form = new FormData()
  form.append('file', file)
  return request.post<Attachment>(`/attachments?entity_type=${entity_type}&entity_id=${entity_id}`, form, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

export function deleteAttachment(id: string) {
  return request.delete(`/attachments/${id}`)
}
